Documentation/special-services/pod-mutator.md

```diff
@@ -0,0 +1,76 @@
+# PodMutator Service
+To provide pod mutation capabilities for finer-grained control over rkt containerization concepts and debugging needs, 
+this proposal introduces a new optional stage1 app/service with special semantics that is lightweight.
```

Comment:
It kinda depends how first-class you want to make container support imo.

If you have a similar scheme on a per-app basis as we currently have on a per-pod basis (locks and states), or so on then you can use attempts to acquire that lock as a means to get status events.

Proactive events vs polling status is not a key distinction in the CRI and, while events are preferred, they're not explicitly required. Hooks (pre/post start/stop), iiuc, are going to be implemented externally in kubelet so they're not related to this.